lib/hal_api/controller/actions.rb

```diff
@@ -80,20 +88,18 @@ def root_resource(resource)
   end
 
   def hal_authorize(resource)
-    if respond_to?(:authorize)
-      authorize(resource)
-    else
-      define_singleton_method(:authorize, Proc.new(resource) do
+    if !respond_to?(:authorize)
+      define_singleton_method(:authorize) do |_resource|
         true
-      end)
+      end
       singleton_class.send(:alias_method, :hal_authorize, :authorize)
     end
+
+    authorize(resource)
```

Comment:
Was the issue here that `hal_authorize` wasn't returning true on the first call, if you didn't explicitly define an `authorize`?  (It looks like it was returning the result of `alias_method` on the first call).